### ISSUE TITLE
Store cards - Add Value after Order closing

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#2372]
_Imported from trac ticket 2372.
Created by **sjohnson**
Opened in trac: Jun 12, 2013, 10:04:19 AM
Last modified in trac: Jul 3, 2013, 8:37:54 PM_
***
_by sjohnson:_
Delay posting value to the store card until the line/order on which it is recorded is closed.  This parallels what happens with other (real) merchandise and would fix the issue of voided card sales.

**Comment by kentbower** [comment]
_**kbower** trac comment on Jun 12, 2013, 10:39:09 AM:_

Auditing is separate issue.
Raising to blocker as this is way too easy for company to lose money.

**Comment by kentbower** [comment]
_**kbower** trac comment on Jun 12, 2013, 10:42:21 AM:_

# Important Implementation Note
It's mandatory that this ticket be committed along with data update sql that will reduce the balance on cards on any open lines so that when they are closed, Loft does not double the credit.
This balance needs to be allowed to go negative during this sql for bookkeeping.

**Comment by kentbower** [comment]
_**kbower** trac comment on Jun 12, 2013, 11:10:46 AM:_

## Note 2
This cannot be done at save time unless not gers_install because many/most orders are finalized with GERS.  This needs to be a background process that processes orders as their status changes to 'F'.

**Comment by kentbower** [comment]
_**kbower** trac comment on Jun 26, 2013, 6:04:22 PM:_

 * Current Processes screen
 * Update event status ability, while locking other fields.
 * update sql (see above)
 * fix [#2395](2395)

**Comment by kentbower** [comment]
_**kbower** trac comment on Jul 3, 2013, 8:37:54 PM:_

(Changeset [[4313]](../commit/82d8e033548d412df996408a556fa8fc73c98b6f))
* Add Process Administration to Flex client (start/stop of Loft processes via supervisor http XML-RPC API) #all
   * Permissions:
     * 'processstop': 'ADMIN - Ability to stop loft processes.'
     * 'processstart':'ADMIN - Ability to start loft processes.'
   * requires configuring [inet_http_server] section in app.ini (see app.ini.template)
 * Support Date with Time display in Data Grids! (not just Date) #all
 * Added ability to Copy (for Paste) from disabled text fields in Flex Client #all
 * fix Store Card loading problem to load store cards only AFTER order is closed (fixes #2372) #all
   * NOTE: existing Store Card balances from open lines will be decreased until line/order is closed
   * new "events.py" Process to process subscriptionevents
   * new subscription for processing orders that have been closed (with trigger, so it doesn't matter whether Loft or GERS closes order)
 * "set_pending" Action for SystemEvent and SubscriptionEvent to reset events to Pending en masse  

 * rework automatic trigger/event generation to support Oracle 8i CLOB
   * fix some JSON problems
 * support drop/readding triggers for postgres (within same transaction), working around lack of CREATE OR REPLACE
 * support dropping unreferenced triggers for postgres
 * subscriptions allow for either standard Loft (method) or rule-driven processing
 * fixes to ProcessBase to capture Exceptions within process of each record instead of waiting for commit flush()
 * formatDateTimeColumn() for DateTime display in datagrid columns
 * Ability to span multiple rows in SpanningTileLayout with 'textheight' for SpanningHGroup

**Comment by kentbower** [comment]
_**kbower** trac status change on Jul 3, 2013, 8:37:54 PM:_ **closed**
